pgmpy/factors/CPD.py

```diff
@@ -125,6 +127,19 @@ def get_cpd(self):
         else:
             return self.values.reshape(1, np.prod(self.cardinality))
 
+    def __str__(self):
+        return self._str(html=False)
+
+    def _str(self, html=False):
+        table_list = []
+        indexes = np.array(list(product(*[range(i) for i in self.cardinality[1:]])))
+        for i in range(1, len(self.cardinality)):
+            row = self.scope()[i]
```

Comment:
In each iteration you are calling the method ```scope()``` which is an overhead. Instead store ```scope = self.scope()``` prior to the loop.